modules/programs/smug.nix

```
@@ -0,0 +1,151 @@
+                    "even-horizontal"
+                    "tiled"
+                  ];
+                  description = ''
+                    Layout of window when opening panes.
+                  '';
+                };
+
+                manual = lib.mkOption {
+                  type = lib.types.nullOr lib.types.bool;
+                  default = null;
+                  description = ''
+                    Start window only manually, using the -w arg
+                  '';
+                };
+
+                panes = lib.mkOption {
+                  default = null;
+                  description = ''
+                    Panes to open in a window.
+                  '';
+                  type = lib.types.nullOr (lib.types.listOf
+                    (lib.types.submodule [{
+                      options = {
+                        root = mkOptionRoot
+                          "Root path of pane. This is relative to the path of the smug project.";
+                        commands = mkOptionCommands
+                          "Commands to execute when pane starts.";
+                        type = lib.mkOption {
+                          type = lib.types.enum [ "horizontal" "vertical" ];
+                          description = ''
+                            Type of pane.
+                          '';
+                        };
+
+                      };
+                    }]));
+                };
+
+              };
+            }]);
+            description = ''
+              Windows to create in the project session
+            '';
+          };
+
+          env = lib.mkOption {
+            type = lib.types.nullOr (lib.types.attrsOf lib.types.str);
+            default = null;
+            description = "Environment Variables to set in session.";
+          };
+          beforeStart = mkOptionCommands
+            "Commands to execute before the tmux-session starts.";
+          stop = mkOptionCommands
+            " Commands to execute after the tmux-session is destroyed.";
+
+        };
+      }
+
+        ]);
+      default = { };
+      description = "List of project configurations.";
+    };
+  };
+
+  config = let
+    cleanedProjects =
+      lib.filterAttrsRecursive (name: value: value != null) cfg.projects;
+
+    mkProjects = lib.attrsets.mapAttrs' (k: v: {
+      name = "${config.home.homeDirectory}/.config/smug/${k}.yml";
+      value.source = let
+        prjConf = lib.attrsets.mapAttrs' (name: value:
+          (lib.attrsets.nameValuePair
+            (if name == "beforeStart" then "before_start" else name) (value))) v
+          // {
+            session = k;
+            windows = lib.lists.forEach v.windows (winprop:
+              (lib.filterAttrsRecursive (name: value: value != null) winprop));
+          };
+      in iniFormat.generate "smug-project-${k}" prjConf;
+    });
+
+  in lib.mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+    home.file = { } // (mkProjects cleanedProjects);
```

```suggestion
    home.file = mkProjects cleanedProjects;
```
Didn't check locally, does this work?